incubator/tensorflow-serving/README.md

```diff
@@ -0,0 +1,159 @@
+
+
+## Copy a Model in NAS Storage
+
+
+*  create `/serving` directory in the NFS server side, take `10.244.1.4` as example
+
+```
+mkdir /nfs
+mount -t nfs -o vers=4.0 10.244.1.4:/ /nfs
+mkdir -p /nfs/serving
+umount /nfs
+```
+
+* Put the mnist model into NAS
+
+```
+mkdir /serving
+mount -t nfs -o vers=4.0 10.244.1.4:/serving /serving
+mkdir -p /serving/model
+cd /serving/model
+curl -O https://github.com/cheyang/charts/raw/features/tensorflow_serving/incubator/tensorflow-serving/models/mnist-export.tar.gz
+tar -xzvf mnist-export.tar.gz
+rm -rf mnist-export.tar.gz
+cd /
+```
+
+* You will see that the contents of the model are stored in the  directory. This is the first version of the model that we will serve.
+
+```
+tree /serving/model/mnist
+/serving/model/mnist
+└── 1
+    ├── saved_model.pb
+    └── variables
+        ├── variables.data-00000-of-00001
+        └── variables.index
+
+umount /serving
+```
+
+## Create Persistent Volume
+
+Creating Persistent Volume with configuration like below
+
+```
+--- 
+apiVersion: v1
+kind: PersistentVolume
+metadata: 
+  labels: 
+    model: mnist
+  name: pv-nas-mnist
+spec:
+  persistentVolumeReclaimPolicy: Recycle
+  accessModes: 
+    - ReadWriteMany
+  capacity: 
+    storage: 5Gi
+  nfs:
+  	# FIXME: use the right IP
+    server: 10.244.1.4
+    path: "/serving/model/mnist"
+```
+
+## Prepare values.yaml
+
+* To deploy with GPU, you can create `values.yaml` like
+
+```
+---
+command: 
+  - /usr/bin/tensorflow_model_server
+args: 
+  - "--port=9090"
+  - "--model_name=mnist"
+  - "--model_base_path=/serving/model/mnist"
+image: "cheyang/tf-model-server-gpu:1.4"
+persistence: 
+  mountPath: /serving/model/mnist
+  pvc: 
+    matchLabels: 
+      model: mnist
+    storage: 5Gi
+```
+
+* To deploy without GPU, you can create `values.yaml` like 
+
+```
+---
+command: 
+  - /usr/bin/tensorflow_model_server
+args: 
+  - "--port=9090"
+  - "--model_name=mnist"
+  - "--model_base_path=/serving/model/mnist"
+image: "cheyang/tf-model-server:1.4"
+persistence: 
+  mountPath: /serving/model/mnist
+  pvc: 
+    matchLabels: 
+      model: mnist
+    storage: 5Gi
+```
+
+## Installing the Chart
+
+To install the chart with the release name `mnist`:
+
+```bash
+$ helm install --values values.yaml --name mnist incubator/acs-tensorflow-serving
+```
+
+## Uninstalling the Chart
+
+To uninstall/delete the `mnist` deployment:
+
+```bash
+$ helm delete mnist
+```
+
+The command removes all the Kubernetes components associated with the chart and
+deletes the release.
+
+## Configuration
+
+The following tables lists the configurable parameters of the Service Tensorflow Serving
+chart and their default values.
+
+| Parameter | Description | Default |
+|-----------|-------------|---------|
+| `image` | TensorFlow Serving image | `registry.cn-hangzhou.aliyuncs.com/tensorflow-samples/tensorflow-serving:1.4.0-devel-gpu` |
```

Comment:
Are the DockerFiles for these images available somewhere? It would be good to link to them so people can build their own images and see how you've done it.